code/game/objects/items/defibrillator.dm

```diff
@@ -69,6 +78,21 @@
 	update_icon()
 
 
+///Wrapper to guarantee powercells are properly nulled and avoid hard deletes.
+/obj/item/defibrillator/proc/set_dcell(obj/item/cell/new_cell)
+	if(dcell)
+		UnregisterSignal(dcell, COMSIG_PARENT_QDELETING)
```

Comment:
shouldnt you qdel the existing cell rather than just leaving it as a referenceless cell or a cell inside the defib's contents.

Reply:
That would be assuming a lot. This proc only deals with changing the variable from one value to another and reacting to the event.
Extra handling of the cell needs to be done elsewhere.